src/components/ErrorBoundary/index.tsx

```diff
@@ -1,66 +1,157 @@
-import React, { PropsWithChildren } from 'react'
+import { ButtonLight, ButtonPrimary } from 'components/Button'
+import { ChevronUpIcon } from 'nft/components/icons'
+import { useIsMobile } from 'nft/hooks'
+import React, { PropsWithChildren, useState } from 'react'
+import { Copy } from 'react-feather'
 import styled from 'styled-components/macro'
 
-import { ExternalLink, ThemedText } from '../../theme'
-import { AutoColumn } from '../Column'
-import { AutoRow } from '../Row'
+import { CopyToClipboard, ExternalLink, ThemedText } from '../../theme'
+import { Column } from '../Column'
 
 const FallbackWrapper = styled.div`
   display: flex;
-  flex-direction: column;
-  width: 100%;
-  align-items: center;
-  z-index: 1;
+  width: 100vw;
+  height: 100vh;
 `
 
 const BodyWrapper = styled.div<{ margin?: string }>`
+  width: min(100%, 500px);
+  margin: auto;
   padding: 1rem;
+`
+
+const SmallButtonPrimary = styled(ButtonPrimary)`
+  width: auto;
+  font-size: 16px;
+  padding: 10px 16px;
+  border-radius: 12px;
+`
+
+const SmallButtonLight = styled(ButtonLight)`
+  font-size: 16px;
+  padding: 10px 16px;
+  border-radius: 12px;
+`
+
+const StretchedRow = styled.div`
+  display: flex;
+  gap: 24px;
+
+  > * {
+    display: flex;
+    flex: 1;
+  }
+`
+
+const Code = styled.code`
+  font-weight: 300;
+  font-size: 12px;
+  line-height: 16px;
+  word-wrap: break-word;
   width: 100%;
+  color: ${({ theme }) => theme.textPrimary};
+  font-family: courier, courier new, serif;
+  overflow: scroll;
+  max-height: calc(100vh - 450px);
+`
+
+const Separator = styled.div`
+  border-bottom: 1px solid ${({ theme }) => theme.backgroundOutline};
 `
 
 const CodeBlockWrapper = styled.div`
-  background: ${({ theme }) => theme.deprecated_bg0};
-  overflow: auto;
-  white-space: pre;
+  display: flex;
+  flex-direction: column;
+  background: ${({ theme }) => theme.backgroundModule};
   box-shadow: 0px 0px 1px rgba(0, 0, 0, 0.01), 0px 4px 8px rgba(0, 0, 0, 0.04), 0px 16px 24px rgba(0, 0, 0, 0.04),
     0px 24px 32px rgba(0, 0, 0, 0.01);
   border-radius: 24px;
-  padding: 18px 24px;
-  color: ${({ theme }) => theme.deprecated_text1};
+  padding: 24px;
+  gap: 10px;
+  color: ${({ theme }) => theme.textPrimary};
+`
+
+const ShowMoreButton = styled.div`
+  display: flex;
+  cursor: pointer;
+  justify-content: space-between;
+`
+
+const CopyIcon = styled(Copy)`
+  stroke: ${({ theme }) => theme.textSecondary};
 `
 
-const Padding = styled.div`
-  padding: 6px 24px;
+const ShowMoreIcon = styled(ChevronUpIcon)<{ $isExpanded?: boolean }>`
+  transform: ${({ $isExpanded }) => ($isExpanded ? 'none' : 'rotate(180deg)')};
 `
 
-const Fallback = ({ error }: { error: Error }) => {
+const CodeTitle = styled.div`
+  display: flex;
+  gap: 14px;
+  align-items: center;
+  word-break: break-word;
+`
+
+const Fallback = ({ error, eventId }: { error: Error; eventId: string | null }) => {
+  const [isExpanded, setExpanded] = useState(false)
+  const isMobile = useIsMobile()
```

Comment:
not a huge fan of this hook. can you just use css for this?

Reply:
Theoretically yes, I am not happy of it either as per my comment few lines later. However, I don't think there's a reasonable and clean way to do it right now while referring to values and names used in Figma. Figma specifies HeadlineSmall/BodySmall and if I were to use media queries, this would require hardcoding values for smaller viewports. 

This approach is generally preferred and considered good practice, it's just that it doesn't play well with the way we do things atm. 

Imagine, if text wasn't a component, but a `css` from Styled Components, we could just include different styles based on media queries, such as:
```
styled.div`
${BodySmall}

@media (min-width: 500px) {
  ${BodyLarge}
 }
 `
 ```
 and problem would be solved!

Reply:
i think we would just display: none if on mobile vs not

Reply:
it's not a huge deal for now, figured i'd mention

Reply:
Oh yes, that works too! Still, I think I'd do it behind a switch to potentially avoid rendering text twice if SEO ever mattered for this page haha